[FLINK-25775] Add Javascript SDK documentation

igalshilman opened the pull request:
This PR adds the JavaScript API to the docs:


index:
<img width="247" alt="image" src="https://user-images.githubusercontent.com/546103/150853862-a48a5747-3951-4d54-8f3e-037013547a09.png">

js:
<img width="1159" alt="image" src="https://user-images.githubusercontent.com/546103/150854026-72267202-f840-46ac-af32-f8cae61a12a8.png">

To build the docs locally use:
```
hugo -b "" serve
```

tillrohrmann commented on docs/content/docs/sdk/js.md:
```diff
@@ -0,0 +1,267 @@
+    typename: "com.example.fns/greeter",
+    fn(context, message) {
+        const name = message.asString();
+        let seen = storage.seen || 0;
```
```suggestion
        let seen = context.storage.seen || 0;
```

tillrohrmann commented on docs/content/docs/sdk/js.md:
```diff
@@ -0,0 +1,267 @@
+        const name = message.asString();
+        let seen = storage.seen || 0;
+        seen = seen + 1;
+        storage.seen = seen;
```
```suggestion
        context.storage.seen = seen;
```

tillrohrmann reviewed (approved):
Thanks for creating this PR @igalshilman. The changes look good to me. +1 for merging after resolving my two minor comments.

igalshilman commented:
Thanks a lot for the review @tillrohrmann ! 
Good catch with the missing `context`! I've made another  pass and merged after addressing your comments!
